client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/NewInteractionEventDialog/PlacesAdditionalForms/TransportationEventForm/TransportationAdditionalForms/FlightEventForm.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { Grid } from '@material-ui/core';
+
+import useFormStyles from 'styles/formStyles';
+import FormInput from 'commons/FormInput/FormInput';
+import CircleTextField from 'commons/CircleTextField/CircleTextField';
+
+import { InteractionEventVariablesConsumer, InteractionEventVariables } from '../../../InteractionEventVariables';
+
+const FlightEventForm : React.FC = () : JSX.Element => {
+
+    const formClasses = useFormStyles();
+
+    const onFlightNumberChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, ctxt: InteractionEventVariables) => 
+        ctxt.setFlightNumber && ctxt.setFlightNumber(event.target.value);
+
+    const onAirlineChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, ctxt: InteractionEventVariables) => 
+        ctxt.setAirline && ctxt.setAirline(event.target.value);
+    
+    const onBoardingCityChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, ctxt: InteractionEventVariables) => 
+        ctxt.setBoardingCity && ctxt.setBoardingCity(event.target.value);
+    
+    const onBoardingCountryChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, ctxt: InteractionEventVariables) => 
+        ctxt.setBoardingCountry && ctxt.setBoardingCountry(event.target.value);
+
+    const onEndCityChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, ctxt: InteractionEventVariables) => 
+        ctxt.setEndCity && ctxt.setEndCity(event.target.value);
+    
+    const onEndCountryChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, ctxt: InteractionEventVariables) => 
+        ctxt.setEndCountry && ctxt.setEndCountry(event.target.value);
+
+    return (
+        <InteractionEventVariablesConsumer>
+            {
+                ctxt =>
```

Comment:
can do useContext instead and save the extra param in the functions

Reply:
same for the other forms tbh

Reply:
but isn't useContext creating a new context?

Reply:
not if you pass an existing context as a prop

Reply:
https://reactjs.org/docs/hooks-reference.html#usecontext